### PR TITLE
fix: core storybook token/theme switcher

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

When switching themes in the core storybook, token snippets would not stay in sync displaying the wrong values. Example, when switching to dark theme in storybook and then viewing the color alias tokens the user would see the light theme hsl values for a dark theme color.

## What is the new behavior?
This PR updates the Storybook version improving the theme switcher to show the correct token values. A side effect of updating is also improved performance with storybook core.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/.storybook/preview.js [hunk 1]
```diff
@@ -105,12 +105,56 @@ export const parameters = {
...
+  //   description: 'Clarity Animations',
+  //   defaultValue: '',
+  //   toolbar: {
+  //     items: [
+  //       { value: '', title: 'Enable Animations' },
+  //       { value: 'low-motion', title: 'Disable Animations' },
+  //     ],
+  //   },
+  // },
+};
```
Comment (coryrylan) [inline packages/core/.storybook/preview.js:143]: Placeholder for when Motion is merged

File: packages/core/.storybook/preview.js [hunk 2]
```diff
@@ -105,12 +105,56 @@ export const parameters = {
...
+
+const themeDecorator = (story, { globals }) => {
+  const themes = `${globals.theme ? globals.theme : ''} ${globals.motion ? globals.motion : ''}`;
+  document.body.setAttribute('cds-theme', `${themes}`);
+  window.parent.document.body.setAttribute('cds-theme', `${themes}`);
+  document.documentElement.setAttribute('cds-base-font', `${globals.baseFont === '100%' ? '16' : ''}`);
+  return story();
+};
+
+export const decorators = [themeDecorator];
```
Comment (coryrylan) [inline packages/core/.storybook/preview.js:153]: Storybook decorator allows the theme to be toggled on the storybook manager UI as well as the demo iframe.

File: packages/core/.storybook/public/demo.css [hunk 3]
```diff
@@ -11,21 +11,44 @@ html {
...
+table tr {
+  border-top-color: var(--cds-alias-object-border-color) !important;
+}
+
+table td button {
+  color: var(--cds-global-typography-color-400) !important;
+  font-weight: 400 !important;
+  background: var(--cds-alias-object-interaction-background) !important;
+}
+
```
Comment (coryrylan) [inline packages/core/.storybook/public/demo.css:43]: Minor style improvements to the embedded storybook tables when theme switching

File: packages/core/docs/angular.stories.mdx [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { html, LitElement } from 'lit-element';
 
 <Meta title="Documentation/Angular" />
 
+<Story id="stories-clarity--theme-switcher" inline={true} />
+
```
Comment (coryrylan) [inline packages/core/docs/angular.stories.mdx:7]: This is a placeholder story for the storybook pages that are plain text content. The decorator theme switcher only runs if there is at least one story on the page.

File: packages/core/docs/utilities.stories.ts [hunk 5]
```diff
@@ -12,7 +12,6 @@ export default {
   title: 'Utilities (Preview)/utils',
   parameters: {
     options: { showPanel: true },
-    a11y: { disable: true }, // disabled for scroll jump
```
Comment (coryrylan) [inline packages/core/docs/utilities.stories.ts:15]: No longer needed in the dependencies

File: packages/core/package.json [hunk 6]
```diff
@@ -62,19 +62,12 @@
...
-    "@storybook/addon-controls": "6.0.21",
-    "@storybook/addon-cssresources": "6.0.21",
-    "@storybook/addon-docs": "6.0.21",
-    "@storybook/addon-knobs": "6.0.21",
-    "@storybook/addon-links": "6.0.21",
-    "@storybook/addon-storysource": "6.0.21",
-    "@storybook/addon-viewport": "6.0.21",
-    "@storybook/addons": "6.0.21",
-    "@storybook/cli": "6.0.21",
-    "@storybook/web-components": "6.0.21",
```
Comment (coryrylan) [inline packages/core/package.json:77]: The addon-essentials package storybook provides combines most of these into a single base package simplifying the number of dependencies needed.

File: packages/core/src/styles/tokens/tokens.stories.mdx [hunk 7]
```diff
@@ -22,17 +22,17 @@ always be prefered over global and component tokens. Component tokens are values
...
-// global token
+/* global token */
 --cds-global-color-green-800 
 
-// alias token
+/* alias token */
 --cds-alias-color-status-success
 
-// component token
+/* component token */
```
Comment (coryrylan) [inline packages/core/src/styles/tokens/tokens.stories.mdx:35]: new version of storybook did not highlight scss as well as css

File: packages/core/src/styles/tokens/tokens.stories.ts [hunk 8]
```diff
@@ -60,20 +64,58 @@ function getTokenTable(key: string) {
...
+    .reduce((prev: any, next: string) => {
+      const rule = next.split(':');
+      return { ...prev, [rule[0].trim()]: rule[1].trim() };
+    }, {});
+}
+
+function createToken(token: Token, globals: any, currentTheme: any) {
+  token.computedValue = getComputedTokenValue(token, globals);
+  token.formattedValue = html`<span cds-text="code">${token.computedValue}</span>`;
+  token.alias = token.alias ? currentTheme[token.name] : undefined;
```
Comment (coryrylan) [inline packages/core/src/styles/tokens/tokens.stories.ts:118]: This addition allows the token to be read from the :root theme active in the browser. This enables the text to render the current value of the token when the theme is switched in storybook.

File: packages/core/src/styles/tokens/tokens.stories.ts [hunk 9]
```diff
@@ -242,7 +286,7 @@ export const spacing = () => getTokenTable('globalSpace');
 
 export const typography = () => getTokenTable('globalTypography');
 
-export const color = () => getTokenTable('globalColor');
+export const color = (_args: any, { globals }: any) => getTokenTable('globalColor', globals);
```
Comment (coryrylan) [inline packages/core/src/styles/tokens/tokens.stories.ts:289]: The args/globals is a storybook api that allows information from the toolbar menu to be passed into stories

**Review by hippee-lee — APPROVED** [review]

**Review by jeeyun — APPROVED** [review]

**Pull request merged**